pages/directory/index.tsx

```diff
@@ -1,28 +1,105 @@
-import { Box, Container, List, ListItem, Typography } from '@mui/material';
+import { Box, Container, CircularProgress, List, ListItem, Typography } from '@mui/material';
+import axios from 'axios';
 import { NextPage } from 'next';
-import React from 'react';
+import * as React from 'react';
 
+import ErrorPopup from 'client/components/ErrorPopup';
 import createAxiosInstance from 'client/lib/axios';
 import { ClientUser } from 'shared/User';
 
-const Directory: NextPage<{ users: ClientUser[]; error: string }> = props => {
-  const { users, error } = props;
+const Directory: NextPage<{ users: ClientUser[]; totalPages: number; error: string }> = props => {
+  const { users, totalPages, error } = props;
+  const lastCardRef = React.useRef(null);
+  const [userData, setUserData] = React.useState(users);
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [totalPageNum, setTotalPageNum] = React.useState(totalPages);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const [open, setOpen] = React.useState(false);
```

Comment:
would prefer to have this be named a little more specifically (if we have more than one alert/modal/etc, what this opens becomes ambiguous).  `showError, setShowError` as a sample alternative